### PR TITLE
[MM-22691][Browserview] fix tray icon

### PR DESCRIPTION
**Pull request by Willyfrog**
**Summary**
fix path for images so tray can load, also fix that on osx theme change it changes as well.

**Issue link**
[MM-22691](https://mattermost.atlassian.net/browse/MM-22691)

**Test Cases**
change settings to display tray menu and restart the app
open osx preferences and switch from dark to light and back

Expected: icon changes to be displayed correctly

**Additional Notes**
requires https://github.com/mattermost/desktop/pull/1387

### CODE REVIEW

File: src/main/main.js [hunk 1]
```diff
@@ -163,13 +162,19 @@ function initializeArgs() {
...
-  WindowManager.setConfig(config.data, config.showTrayIcon, deeplinkingUrl);
+async function initializeConfig() {
+  const loadConfig = new Promise((resolve, reject) => {
+    config = new Config(app.getPath('userData') + '/config.json');
+    config.once('update', (configData) => {
+      config.on('update', handleConfigUpdate);
+      config.on('synchronize', handleConfigSynchronize);
+      config.on('darkModeChange', handleDarkModeChange);
+      handleConfigUpdate(configData);
+      resolve();
```
Comment (Willyfrog) [inline src/main/main.js:173]: in windows, there is a bigger delay until the registry is read, and this causes the config to not be loaded in time, making other parts fail

Comment (devinbinnie) [reply src/main/main.js:173]: Do we need the second `await` in this function? Couldn't we just return the promise and the `await` in `initialize()` will take care of it?

File: webpack.config.renderer.js [hunk 2]
```diff
@@ -42,6 +43,19 @@ module.exports = merge(base, {
       ignoreOrder: true,
       chunkFilename: '[id].[contenthash].css',
     }),
+    new CopyPlugin({
+      patterns: [{
+        from: 'assets/windows/*.ico',
+        context: 'src',
+      }, {
+        from: 'assets/linux/*/*.png',
```
Comment (devinbinnie) [inline webpack.config.renderer.js:51]: I think you can reduce this by doing:
```suggestion
        from: 'assets/**/*.png',
```

**Review by devinbinnie — APPROVED** [review]

**Review by nevyangelova — APPROVED** [review]

File: src/main/main.js (hunk 1 above)
Comment (Willyfrog) [reply src/main/main.js:173]: config is being used by `initializeBeforeAppReady` so we have to wait for it to finish. In windows it needs to go to the registry and get the info from there and it ends up being a race condition, so it feel safer this way.

**Pull request merged**